--- md.f90	2011-05-06 19:16:45.004721725 +0200
+++ md.f90.fix	2011-05-06 19:13:54.410544757 +0200
@@ -14137,6 +14137,12 @@
         EQ(istate)%q%angle = EQ(istate)%q%angle + Eurey*gamma
         du = gamma*2*(qanglib(ic)%ureyfk*ru/dik)*EQ(istate)%lambda
 
+        d(k3+1) = d(k3+1) + du*rik(1)
+        d(k3+2) = d(k3+2) + du*rik(2)
+        d(k3+3) = d(k3+3) + du*rik(3)
+        d(i3+1) = d(i3+1) - du*rik(1)
+        d(i3+2) = d(i3+2) - du*rik(2)
+        d(i3+3) = d(i3+3) - du*rik(3)
 
 if ( icoupl .ne. 0 ) then
 
