--- md.f90	2011-04-27 15:52:57.052588692 +0200
+++ md.f90.u-b_fix	2011-04-27 15:52:30.863335894 +0200
@@ -941,7 +941,7 @@
 j3=j*3-3
 k3=k*3-3
         ! 1-3 distance for Urey-Bradley potential:
-    if(anglib(ic)%ureyfk > 0.) then
+    if(anglib(ic)%ureyfk /= 0.) then
                 rik(1) = x(k3+1) - x(i3+1)
                 rik(2) = x(k3+2) - x(i3+2)
                 rik(3) = x(k3+3) - x(i3+3)
